simple_rest_client/api.py

```diff
@@ -35,24 +37,24 @@ def add_resource(
         append_slash=False,
         json_encode_body=False,
     ):
+        resource_valid_name = self.correct_attribute_name(resource_name)
         resource_class = resource_class or Resource
         resource = resource_class(
             api_root_url=api_root_url or self.api_root_url,
-            resource_name=resource_name,
+            resource_name=resource_valid_name,
             params=params or self.params,
             headers=headers or self.headers,
             timeout=timeout or self.timeout,
             append_slash=append_slash or self.append_slash,
             json_encode_body=json_encode_body or self.json_encode_body,
             ssl_verify=self.ssl_verify,
         )
-        valid_name = self.correct_attribute_name(resource_name)
-        self._resources[valid_name] = resource
-        setattr(self, valid_name, resource)
+        self._resources[resource_valid_name] = resource
```

Comment:
It seems to me that the `_resources` dictionary should have the original `resource_name` as key, rather than the fixed resource name.

```suggestion
        self._resources[resource_name] = resource
```

(though actually this line change happened in a previous commit: #28).